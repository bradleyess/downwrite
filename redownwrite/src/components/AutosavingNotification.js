import React from 'react'
import Toast from './Toast'

/* TypeError: cyclic object value */
export default () => <Toast>Autosaving</Toast>
