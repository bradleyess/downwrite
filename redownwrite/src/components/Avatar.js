import React, { Component } from 'react'
import styled from 'styled-components'
import color from 'rcolor'

const initialState = {
  a: color() || `#FEB692`,
  b: color() || `#EA5455`
}

class RandomColor extends Component {
  state = initialState

  render() {
    return this.props.children(this.state)
  }
}

const Avatar = styled.div`
  border-radius: 3rem;
  height: 3rem;
  width: 3rem;
  background: linear-gradient(
    135deg,
    ${props => props.colors.a || '#FEB692'} 10%,
    ${props => props.colors.b || '#EA5455'} 100%
  );
  margin: 0 auto 1rem;
`

export default () => <RandomColor>{colors => <Avatar colors={colors} />}</RandomColor>
