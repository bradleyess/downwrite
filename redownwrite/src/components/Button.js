// @flow

import styled from 'styled-components'
import { colors } from '../utils/defaultStyles'

export default styled.button`
  display: block;
  font-size: 14px;
  background-color: ${colors.yellow700};
  transition: background-color 250ms ease-in-out;
  color: #282828;
  font-weight: 700;
  border: 0;
  padding: 0.25rem 1.125rem;
  border-radius: 0.25rem;
  box-shadow: 0 0 4px rgba(0, 0, 0, 0.037), 0 4px 8px rgba(0, 0, 0, 0.07);
  &:hover {
    background-color: ${colors.yellow500};
  }

  &:[disabled] {
    filter: grayscale(100%);
  }
`
