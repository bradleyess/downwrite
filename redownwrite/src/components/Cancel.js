import styled from 'styled-components'
import { colors } from '../utils/defaultStyles'

export default styled.button`
  margin-right: 1rem;
  color: ${colors.blue700};
  background: none;
  border: 0px;
  font-weight: 700;
`
