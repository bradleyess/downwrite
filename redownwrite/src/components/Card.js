import React from 'react'
import styled from 'styled-components'
import Link from 'react-router-dom/Link'
import distance from 'date-fns/distance_in_words_to_now'
import { colors } from '../utils/defaultStyles'

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 192px;
  font-weight: 400;
  box-shadow: 0 0 2px rgba(0, 0, 0, 0.07), 0 2px 4px rgba(0, 0, 0, 0.12);
  background-color: white;
`

const CardTitle = styled.h2`
  font-size: 14px;
  margin-bottom: 0px;
  font-weight: 700;

  @media (min-width: 57.75rem) {
    font-size: 18px;
  }
`

const CardHeader = styled.header`
  display: block;
  border-bottom: 1px solid ${colors.gray200};
  padding: 12px 8px;
`

const CardMeta = styled.small`
  font-size: 12px;
  display: block;
  color: ${colors.gray300};
  font-weight: 400;
`

const CardTray = styled.footer`
  display: flex;
  padding: 8px;
  font-weight: 700;
  font-size: 12px;
  justify-content: space-between;
  background-color: rgba(101, 163, 191, 0.125);

  & a:hover {
    color: ${colors.blue500};
  }
`

const CardPreview = styled.p`
  font-size: 12px;
`

const CardBody = styled.div`
  display: block;
  padding: 0.5rem;
  flex: 1;
  overflow: hidden;
`

const CardDelete = styled.button`
  color: ${colors.blue700};
  border: 0px;
  background: none;
  appearance: none;
  -webkit-font-smoothing: antialiased;
`

const SxLink = styled(Link)`
  margin-right: 8px;
`

const Card = ({ title, id, content, dateAdded, onDelete, ...args }) => (
  <CardContainer data-testid="CARD">
    <CardHeader>
      <CardTitle data-testid="CARD_TITLE">
        <Link to={`/${id}/edit`}>{title}</Link>
      </CardTitle>
      <CardMeta>added {distance(dateAdded)} ago</CardMeta>
    </CardHeader>

    <CardBody>
      {content && (
        <CardPreview data-test="snippet">{content.blocks[0].text.substr(0, 75)}</CardPreview>
      )}
    </CardBody>
    <CardTray>
      <div data-testid="CARD_EXCERPT">
        <SxLink to={`/${id}/edit`}>Edit</SxLink>
        {args.public && <Link to={`/${id}/preview`}>Preview</Link>}
      </div>
      {onDelete && (
        <CardDelete data-testid="CARD_DELETE_BUTTON" onClick={onDelete}>
          Delete
        </CardDelete>
      )}
    </CardTray>
  </CardContainer>
)

export default Card
