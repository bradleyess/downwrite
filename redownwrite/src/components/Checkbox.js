import React from 'react'
import { css } from 'glamor'
import icon from './check.svg'

const sharedStyle = css({
  display: 'inline-block',
  borderRadius: 4,
  verticalAlign: 'middle',
  width: 20,
  height: 20,
  appearance: 'none',
  border: 0
})

const checkedStyle = css(sharedStyle, {
  background: `var(--color-1) url(${icon}) no-repeat center center`,
  border: 0
})

const unCheckedStyle = css(sharedStyle, { background: '#D0D0D0' })

export default props => (
  <input
    type="checkbox"
    className={css(props.checked ? checkedStyle : unCheckedStyle)}
    {...props}
  />
)
