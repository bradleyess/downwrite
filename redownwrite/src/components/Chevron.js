import React from 'react'

export default ({ open }: { open: boolean }) => (
  <svg
    className="Chevron"
    viewBox="0 0 16 16"
    width="16"
    height="16"
    fill="none"
    stroke="currentColor"
    strokeWidth={2}>
    <path
      d={
        open
          ? 'M1.0606601717798212 11 L8 4.060660171779821 L14.939339828220179 11'
          : 'M1.0606601717798212 5 L8 11.939339828220179 L14.939339828220179 5'
      }
    />
  </svg>
)
