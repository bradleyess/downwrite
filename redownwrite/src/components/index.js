export { default as Button } from './Button'
export { default as Cancel } from './Cancel'
export { default as Card } from './Card'
export { default as DWEditor } from './Draft'
export { default as NightMode } from './NightMode'
export { default as EmptyPosts } from './EmptyPosts'
export { default as Header } from './Header'
export { default as Input } from './Input'
export { default as Loading } from './Loading'
export { default as Logo } from './Logo'
export { default as Nav } from './Nav'
export { default as PostList } from './PostList'
export { default as Toolbar } from './Toolbar'
export { default as Wrapper } from './Wrapper'
export { default as Helpers } from './Helpers'
export { default as LoginInput } from './LoginInput'
export { default as LoginButton } from './LoginButton'
export { default as SettingsIcon } from './SettingsIcon'
export { default as Layout } from './Layout'
export { default as Modal } from './Modal'
export { default as Toggle } from './Toggle'
export { default as Export } from './Export'
export { default as Privacy } from './Privacy'
export { default as Check } from './Checkbox'
export { default as CloseIcon } from './CloseIcon'
export { default as InvalidToken } from './InvalidToken'
export { default as Content } from './Content'
export { default as Logger } from './Logger'
export { default as UIFlash } from './UIFlashMessages'
