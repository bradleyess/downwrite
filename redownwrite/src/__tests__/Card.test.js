import { Card } from '../components'
import { posts } from './db.json'

const mockDelete = jest.fn()
const id = '6acebce0-20b6-4015-87fe-951c7bb36481'

jest.mock('react-router-dom/Link')

const wrapper = mount(
  <Card title="Starting Again" content={posts[0].content} id={id} onDelete={mockDelete} />
)

describe('Card', () => {
  it('contains snippet from content', () => {
    const snippet = wrapper.find('[data-test="cardDelete"]').text()
    expect(snippet.length).toBeLessThanOrEqual(90)
  })

  it('contains a title', () => {
    console.log(wrapper.find('Link'))
    expect(
      wrapper
        .find('Link')
        .first()
        .text()
    ).toBe('Starting Again')
  })

  // TODO: snapshots are going to be fucked for a while until
  // converted to next.js
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should fire a delete', () => {
    wrapper.find('[data-test="cardDelete"]').simulate('click')

    expect(mockDelete).toHaveBeenCalled()
  })
})
