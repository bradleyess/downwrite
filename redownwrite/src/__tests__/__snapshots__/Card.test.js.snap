// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot 1`] = `
.glamor-0,
[data-glamor-0] {
  font-size: 14px;
  margin-bottom: 0;
  font-weight: 700;
}

.glamor-5,
[data-glamor-5] {
  color: var(--color-2);
  border: 0;
  background: none;
  appearance: none;
  -webkit-font-smoothing: antialiased;
  -webkit-appearance: none;
  -moz-appearance: none;
}

.glamor-2,
[data-glamor-2] {
  font-size: 12px;
}

.glamor-9,
[data-glamor-9] {
  font-weight: 400;
  box-shadow: 0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12);
  background-color: white;
}

.glamor-1,
[data-glamor-1] {
  font-size: 12px;
  display: block;
  color: #757575;
  font-weight: 400;
}

.glamor-6 a,
[data-glamor-6] a {
  color: #4382A1;
}

.glamor-3,
[data-glamor-3] {
  margin-right: 8px;
}

.css-m8rrnj,
[data-css-m8rrnj] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  height: 192px;
  flex-direction: column;
  justify-content: space-between;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.css-usmh8e,
[data-css-usmh8e] {
  display: block;
  border-bottom: 1px solid #DBDCDD;
  padding: 12px 8px;
}

.css-p2oddx,
[data-css-p2oddx] {
  display: block;
  padding: 8px;
  flex: 1;
  overflow: hidden;
  -webkit-flex: 1;
}

.css-chikq1,
[data-css-chikq1] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  padding: 8px;
  font-weight: 700;
  font-size: 12px;
  justify-content: space-between;
  background-color: rgba(101, 163, 191, .125);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.css-fisw11,
[data-css-fisw11] {
  display: block;
}

@media (min-width: 57.75rem) {
  .glamor-0,
  [data-glamor-0] {
    font-size: 18px;
  }
}

<Card
  content={
    Object {
      "blocks": Array [
        Object {
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "er5rb",
          "text": "What am i really going to do here?",
          "type": "unstyled",
        },
        Object {
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "ds3fg",
          "text": "",
          "type": "unstyled",
        },
        Object {
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "9sgm6",
          "text": "I am trying to write a thing",
          "type": "unstyled",
        },
        Object {
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "a0t8m",
          "text": "",
          "type": "unstyled",
        },
        Object {
          "data": Object {},
          "depth": 0,
          "entityRanges": Array [],
          "inlineStyleRanges": Array [],
          "key": "2ok0",
          "text": "With many lines",
          "type": "unstyled",
        },
      ],
      "entityMap": Object {},
    }
  }
  id="6acebce0-20b6-4015-87fe-951c7bb36481"
  onDelete={[Function]}
  title="Starting Again"
>
  <Flex
    className={
      Object {
        "data-glamor-9": "",
      }
    }
    data-test="Card"
    flexDirection="column"
    height={192}
    justifyContent="space-between"
  >
    <View
      className={
        Object {
          "data-glamor-9": "",
        }
      }
      component="div"
      data-test="Card"
      display="flex"
      flexDirection="column"
      height={192}
      justifyContent="space-between"
    >
      <div
        className={
          Object {
            "data-glamor-9": "",
          }
        }
        data-css-m8rrnj=""
        data-test="Card"
      >
        <Block
          borderBottom="1px solid #DBDCDD"
          padding="12px 8px"
        >
          <View
            borderBottom="1px solid #DBDCDD"
            component="div"
            display="block"
            padding="12px 8px"
          >
            <div
              data-css-usmh8e=""
            >
              <h2
                className={
                  Object {
                    "data-glamor-0": "",
                  }
                }
              >
                <Link
                  to="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                >
                  <a
                    href="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                  >
                    Starting Again
                  </a>
                </Link>
              </h2>
              <small
                className={
                  Object {
                    "data-glamor-1": "",
                  }
                }
              >
                added 
                almost NaN years
                 ago
              </small>
            </div>
          </View>
        </Block>
        <Block
          flex={1}
          overflow="hidden"
          padding={8}
        >
          <View
            component="div"
            display="block"
            flex={1}
            overflow="hidden"
            padding={8}
          >
            <div
              data-css-p2oddx=""
            >
              <p
                className={
                  Object {
                    "data-glamor-2": "",
                  }
                }
                data-test="snippet"
              >
                What am i really going to do here?
              </p>
            </div>
          </View>
        </Block>
        <Flex
          backgroundColor="rgba(101, 163, 191, .125)"
          className={
            Object {
              "data-glamor-6": "",
            }
          }
          fontSize={12}
          fontWeight={700}
          justifyContent="space-between"
          padding={8}
        >
          <View
            backgroundColor="rgba(101, 163, 191, .125)"
            className={
              Object {
                "data-glamor-6": "",
              }
            }
            component="div"
            display="flex"
            fontSize={12}
            fontWeight={700}
            justifyContent="space-between"
            padding={8}
          >
            <div
              className={
                Object {
                  "data-glamor-6": "",
                }
              }
              data-css-chikq1=""
            >
              <Block>
                <View
                  component="div"
                  display="block"
                >
                  <div
                    data-css-fisw11=""
                  >
                    <Link
                      className={
                        Object {
                          "data-glamor-3": "",
                        }
                      }
                      to="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                    >
                      <a
                        className={
                          Object {
                            "data-glamor-3": "",
                          }
                        }
                        href="/6acebce0-20b6-4015-87fe-951c7bb36481/edit"
                      >
                        Edit
                      </a>
                    </Link>
                  </div>
                </View>
              </Block>
              <button
                className={
                  Object {
                    "data-glamor-5": "",
                  }
                }
                data-test="cardDelete"
                onClick={[Function]}
              >
                Delete
              </button>
            </div>
          </View>
        </Flex>
      </div>
    </View>
  </Flex>
</Card>
`;
